starter-repos/app-template/validation/functions/kubeval-validator.yaml

```diff
@@ -0,0 +1,32 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: examples.config.kubernetes.io/v1beta1
+kind: Kubeval
+metadata:
+  name: kubeval-validator
+  annotations:
+    config.kubernetes.io/function: |
+      container:
+        image: gcr.io/kustomize-functions/example-validator-kubeval:v0.1.0
+# Please update the following options when necessary
+spec:
+  # Skip validation for resource definitions without a schema. If omitted, a default value of false will be assumed.
+  ignoreMissingSchemas: true
+  # KubernetesVersion is the version of Kubernetes to validate against (default "master").
+  kubernetesVersion: 1.16.0
```

Comment:
What does `master` stand for? 

Ideally this should match the cluster's k8s version.

Reply:
It is the kubernetes version for which to load the schema: https://github.com/instrumenta/kubeval/blob/master/kubeval/config.go#L25.

e.g.:
https://kubernetesjsonschema.dev/master/deployment-apps-v1.json vs. https://kubernetesjsonschema.dev/v1.16.0/deployment-apps-v1.json

Terraform sets up the cluster with release channel, not a specific version: https://github.com/GoogleCloudPlatform/solutions-modern-cicd-anthos/blob/master/1_clusters/clusters.tf#L76.
So it might be tricky to get the actual cluster's k8s version.

Reply:
I see. This is fine for now. We can ask the users to configure it when needed.